be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/validators/ElectionValidator.scala

```diff
@@ -69,7 +88,19 @@ object ElectionValidator extends MessageDataContentValidator with EventValidator
     def validationError(reason: String): PipelineError = super.validationError(reason, "ResultElection", rpcMessage.id)
 
     rpcMessage.getParamsMessage match {
-      case Some(_) => Left(rpcMessage)
+      case Some(message) => 
+        val data: ResultElection = message.decodedData.get.asInstanceOf[ResultElection]
+
+        val sender: PublicKey = message.sender
+        val channel: Channel = rpcMessage.getParamsChannel
+
+        if (!validateOwner(sender, channel)){
+          Right(validationError(s"invalid sender $sender"))
+        } else if (!validateChannelType(ObjectType.ELECTION, channel)) {
+          Right(validationError(s"trying to send a CastVoteElection message on a wrong type of channel $channel"))
```

Comment:
Did you mean "trying to send a ResultElection message on a wrong type of channel $channel"